### ISSUE TITLE
Add ability to extend console app with a plugin layer

### ISSUE THREAD

**Issue by mnlbox** [hasura/graphql-engine#2373]
Hi guys,
I think console app is a very useful and handy thing for many project but now we have many feature request about console app.
If Hasura core team provide a plugin layer around console app then I think community can done many of console related issues as a plugin.
Something like WordPress plugin capability that anyone can easily create and install custom plugin on it without need to fork all of core repository and with a minimum learning curve.
Are you any plan or roadmap about this type of enhancement?
Is it in your priority?

@shahidhk, @tirumaraiselvan, @coco98

**Comment by tirumaraiselvan** [comment]
cc: @rikinsk @dsandip 

I like the idea. @hmexx also suggested to have a component integration for console triggers so they can render some nice UI based on event trigger response instead of the json.

**Comment by hmexx** [comment]
Yes. That was my vision of the MVP version of extensible console.  Manual
Trigger —> Result with HTML (and maybe JS) —> Component rendered in
Console.

More than half the work is done by building out the manual trigger feature.

**Comment by mnlbox** [comment]
@hmexx so you are working on this now?

**Comment by hmexx** [comment]
No, I am not. I’m afraid I don’t have time to contribute at the moment.
